Section 4.3

aleno4ka1998 opened the pull request:
Классы

aleno4ka1998 committed:
Create most distant point.py

aleno4ka1998 reviewed (commented):
здесь я затупила немножечко

aleno4ka1998 committed:
Наиболее удаленная точка

Выведите координаты наиболее удаленной от начала координат точки.

Входные данные
Программа получает на вход набор точек на плоскости. Сначала задано количество точек n, затем идет последовательность из n строк, каждая из которых содержит два числа: координаты точки. Величина n не превосходит 100, все исходные координаты – целые числа, не превосходящие 103 по абсолютной величине.

Выходные данные
Выведите  координаты точки, наиболее удаленной от начала координат.

aleno4ka1998 committed:
Центр тяжести

Выведите координаты центра тяжести данного множества точек.

Создайте структуру Point и сохраните исходные данные в массиве структур Point.

Входные данные
Программа получает на вход набор точек на плоскости. Сначала задано количество точек n, затем идет последовательность из n строк, каждая из которых содержит два числа: координаты точки. Величина n не превосходит 100, все исходные координаты – целые числа, не превосходящие 103.

Выходные данные
Выведите  координаты центра тяжести данного множества точек. Ответ необходимо выводить с точностью в 15 значащих цифр.

aleno4ka1998 committed:
Диаметр множества

Выведите диаметр данного множества – максимальное расстояние между любыми двумя точками.

Создайте структуру Point и сохраните исходные данные в массиве структур Point.

Входные данные
Программа получает на вход набор точек на плоскости. Сначала задано количество точек n, затем идет последовательность из n строк, каждая из которых содержит два числа: координаты точки. Величина n не превосходит 100000, все исходные координаты – целые числа, не превосходящие 103.

Выходные данные
Необходимо вывести  диаметр данного множества с точностью в 15 значащих цифр.

aleno4ka1998 committed:
upd Диаметр множества

досадная ошибка

aleno4ka1998 committed:
Сортировка точек

Выведите все исходные точки в порядке возрастания их расстояний от начала координат.

Создайте структуру Point и сохраните исходные данные в массиве структур Point.

Входные данные
Программа получает на вход набор точек на плоскости. Сначала задано количество точек n, затем идет последовательность из n строк, каждая из которых содержит два числа: координаты точки. Величина n не превосходит 100, все исходные координаты – целые числа, не превосходящие 103.

Выходные данные
Необходимо вывести  все исходные точки в порядке возрастания их расстояний от начала координат. Программа выводит только координаты точек, их количество выводить не надо.

aleno4ka1998 committed:
Максиальный периметр

Среди исходных точек найдите три, образующие треугольник с максимальным периметром. Выведите данный периметр.

Входные данные
Программа получает на вход набор точек на плоскости. Сначала задано количество точек n (2<n<101), затем идет последовательность из n строк, каждая из которых содержит два числа: координаты точки. Все исходные координаты – целые числа, не превосходящие 103.

Выходные данные
Необходимо вывести  найденный периметр с точностью в 15 значащих цифр.

aleno4ka1998 committed:
Максимальная площадь

Среди исходных точек найдите три, образующие треугольник максимальной площади. Выведите данную площадь.

Создайте структуру Point и сохраните исходные данные в массиве структур Point.

Входные данные
Программа получает на вход набор точек на плоскости. Сначала задано количество точек n (2<n<101), затем идет последовательность из n строк, каждая из которых содержит два числа: координаты точки. Все исходные координаты – целые числа, не превосходящие 10 3.

Выходные данные
Необходимо вывести найденную площадь с точностью в 15 значащих цифр.

aleno4ka1998 committed:
Средний балл по предметам

Определите средний балл всех учащихся по каждому предмету.

Входные данные
Заданы сначала количество учащихся n, затем n строк, каждая из которых содержит фамилию, имя и три числа (оценки по трем предметам: математике, физике, информатике). Данные в строке разделены одним пробелом. Оценки принимают значение от 1 до 5.

Выходные данные
Выведите три действительных числа: средний балл всех учащихся по математике, по физике, по информатике.

aleno4ka1998 committed:
Учащиеся без троек

Выведите фамилии и имена учащихся, не имеющих троек (а также двоек и колов).

Входные данные
Заданы сначала количество учащихся n, затем n строк, каждая из которых содержит фамилию, имя и три числа (оценки по трем предметам: математике, физике, информатике). Данные в строке разделены одним пробелом. Оценки принимают значение от 1 до 5.

Выходные данные
Необходимо вывести пары фамилия-имя по одной на строке, разделяя фамилию и имя одним пробелом. Выводить оценки не нужно. Порядок вывода должен быть таким же, как в исходных данных.

thelightningseas commented on section4.3/average score in subjects.py:
```diff
@@ -0,0 +1,32 @@
+class Coordinate:
+    def __init__(self, surname, name, math, physics, it):
+        self.a = surname
```
с такими названиями совершенно непонятно что это за поля. с таким интерфейсом проще использовать список вместо класса)

thelightningseas replied:
возможно если бы сущностью (классом) в этой задаче был не школьник а предмет, было бы проще и понеятнее

thelightningseas commented on section4.3/maximum area.py:
```diff
@@ -0,0 +1,31 @@
+import math
+
+class SetOfCoordinates:
```
если вместо метода который просто аппендит, сделать метод, который сразу обновляет значение максимума, было бы понятнее

человеку, используюещему класс хочется как раз этого - все действия происходят где-то под капотом, а он только использует понятный интерфейс

aleno4ka1998 committed:
Лучшие учащиеся

Определите учащихся с наилучшей успеваемостью, то есть с максимальным средним баллом по трем предметам. Выведите всех учащихся, имеющих максимальный средний балл.

Входные данные
Заданы сначала количество учащихся n, затем n строк, каждая из которых содержит фамилию, имя и три числа (оценки по трем предметам: математике, физике, информатике). Данные в строке разделены одним пробелом. Оценки принимают значение от 1 до 5.

Выходные данные
Необходимо вывести пары фамилия-имя по одной на строке, разделяя фамилию и имя одним пробелом. Выводить оценки не нужно. Порядок вывода должен быть таким же, как в исходных данных.

thelightningseas commented on section4.3/best students.py:
```diff
@@ -0,0 +1,31 @@
+class Pupil:
+
+    def __init__(self, surname, name, math, physics, it):
+        self.a = surname
```
с такими названиями полей никто кроме тебя не сможет пользоваться этим кассом

aleno4ka1998 replied:
Ты про a, b, c и т.д. в ините?

thelightningseas replied:
Да

thelightningseas replied:
![image](https://user-images.githubusercontent.com/46321152/103042535-04e7c180-458b-11eb-93a5-aa02c87869e1.jpeg)

aleno4ka1998 committed:
upd Лучший ученик

thelightningseas commented on section4.3/best students.py:
```diff
@@ -0,0 +1,31 @@
+        maximum = el.averageMark()
+        B.clear()
+        B.append(el)
+print('\n'.join(map(str, B)))
```
мне кажется, понятнее было бы 
```
for pupil in B:
    print(pupil)
```

aleno4ka1998 committed:
Трое лучших

Определите трех учащихся с наилучшим средним баллом по трем предметам. Выведите фамилии и имена этих учащихся. Если при этом у нескольких учащихся средний балл совпадает со средним баллом учащегося, "занявшего 3-е место", то необходимо вывести их всех.

Входные данные
Заданы сначала количество учащихся n, затем n строк, каждая из которых содержит фамилию, имя и три числа (оценки по трем предметам: математике, физике, информатике). Данные в строке разделены одним пробелом. Оценки принимают значение от 1 до 5.

Выходные данные
Необходимо вывести пары фамилия-имя по одной на строке, разделяя фамилию и имя одним пробелом. Выводить оценки не нужно. Порядок вывода должен быть таким же, как в исходных данных.

aleno4ka1998 committed:
Отсортировать по среднему баллу

Выведите фамилии и имена учащихся в порядке убывания их среднего балла.

Входные данные
Заданы сначала количество учащихся n, затем n строк, каждая из которых содержит фамилию, имя и три числа (оценки по трем предметам: математике, физике, информатике). Данные в строке разделены одним пробелом. Оценки принимают значение от 1 до 5.

Общее число учащихся не превосходит 100001.
Выходные данные
Необходимо вывести пары фамилия-имя по одной на строке, разделяя фамилию и имя одним пробелом. Выводить оценки не нужно. Если несколько учащихся имеют одинаковые средние баллы, то их нужно выводить в порядке, заданном во входных данных.

aleno4ka1998 committed:
Личные дела

Однажды, неловкая секретарша перепутала личные дела учащихся. Теперь их снова необходимо упорядочить сначала по классам, а внутри класса по фамилиям

Входные данные
В первой строке дано число N (1 ≤ N ≤ 1000) – количество личных дел. Далее для каждого из N учащихся следующие данные (каждое в своей строке): фамилия и имя, класс, дата рождения. Фамилия и имя – строки не более чем из 20 символов, класс – строка состоящая из числа (от 1 до 11) и латинской буквы (от "A" до "Z" ), дата рождения – дата в формате "ДД.ММ.ГГ" . Гарантируется, что внутри одного класса нет однофамильцев.

Выходные данные
В выходной файл требуется вывести N строк, в каждой из которых записаны данные по одному учащемуся. Строки должны быть упорядочены сначала по классам, а затем по фамилиям.

thelightningseas commented on section4.3/personal documents.py:
```diff
@@ -0,0 +1,22 @@
+n = int(input())
+A = []
+for _ in range(n):
+    surname = str(input())
```
не надо делать str(input()), input() и так возрващает строку, так приходилось делать когда из этой строки хотелось получить число
